### ISSUE TITLE
Undefined property, Undefined variable, and Undefined offset errors

### ISSUE THREAD

**Issue by jkudebeh** [uweb/uw-2014#27]
We are having issues converting to the boundless theme from the older UW theme. We are seeing many errors in the apache error log as well as some pages failing to load, the header is displayed but there is no content.

To eliminate as many variables as possible we performed the following:

installed clean copy of wordpress

in the wp-config.php file changed
define('WP_DEBUG', FALSE);
to
define('WP_DEBUG', TRUE);

Refreshed page, checked apache error log. No errors

added Boundless theme, activated boundless theme, refreshed page.

the followinig errors were logged in the apache error log

PHP Notice:  Undefined property: UW_QuickLinks::$items in /data/www/wp/wp-content/themes/uw-boundless/setup/class.uw-quicklinks.php on line 52, referer: http://192.168.50.4/
PHP Notice:  Undefined variable: menu in /data/www/wp/wp-content/themes/uw-boundless/setup/class.uw-quicklinks.php on line 64, referer: http://192.168.50.4/

activated the Wordpress 2015 theme. No errors recorded in the apache error log file

Activated the Boundless theme
added 4 pages, one home page three child pages, navigated to a child page

the following errors were logged in the apache error log

Undefined offset: 0 in /data/www/wp/wp-content/themes/uw-boundless/page.php on line 10
PHP Notice:  Undefined offset: 0 in /data/www/wp/wp-content/themes/uw-boundless/page.php on line 10, referer: http://192.168.50.4/index.php/home-page/child-page-1/
PHP Notice:  Undefined offset: 0 in /data/www/wp/wp-content/themes/uw-boundless/setup/class.uw-sidebar-menu-walker.php on line 68, referer: http://192.168.50.4/index.php/home-page/child-page-1/
PHP Notice:  Undefined offset: 0 in /data/www/wp/wp-content/themes/uw-boundless/page.php on line 44, referer: http://192.168.50.4/index.php/home-page/

activated the Wordpress 2015 theme. No errors recorded in the apache error log file

tested WordPress 4.2.4 and Wordpress 4.4

I'm showing only the root cause of the error, each error is 8 to 15 lines

**Comment by Nogbit** [comment]
@kilianf Did you or your team get a chance to look at these issue?

**Comment by nwinkelbauer** [comment]
Hi, 

Thank you for bringing these issues to our attention. Sorry for the late response, I must have missed the notification the first time this came in so thank you @Nogbit for pinging again. I will move this issue to the top of my to-do list so hopefully it will be solved by the end of the week. I will keep you all updated as I go through this. 

Thanks, 
Nick

**Comment by jkudebeh** [comment]
Thanks Nick.

I have traced the pages not loading to a function,the_blogroll_banner_style, that was in the UW-Theme and removed in the current theme. I have removed the call to this function.

Looking forward to the fix for the errors in the log file!

thanks
jim

**Comment by nwinkelbauer** [comment]
Hi Guys, 

I have addressed the errors in Commit 9fc3ec74ee78dfa8b881856ae1da2d71e915d03f 

All the errors come from features I added and me not checking to make sure a default value was set. I have fixed these issues so we should be good. I pushed the errors to github as well as live to the cms. 

Let me know whether these worked on your end, just to be sure. 

Thank you for taking the time to dive into the error log and find the root of the problems. It made my job a lot easier! 

Cheers, 
Nick
